functions/configfile.pp

```diff
@@ -0,0 +1,10 @@
+# Generates configuration files for mcollective. Keys are sorted alphabetically:
+# # Configuration is managaged by puppet class main
+# key = valze
+function mcollective::configfile(Hash $confighash) >> String {
+  $sorted = $confighash.keys.sort
+  $result = $sorted.map |$key| {
+    "${key} = ${confighash[$key]}"
+  }
+  (["# Configuration is managed by puppet class ${name}"] << $result << []).join("\n")
```

Comment:
yeah this is broken, lets just remove the header - half managing machines is a smell and thats the only reason anyone wants headers


```puppet
$result = $confighash.keys.sort.map |$key| {
  sprintf("%s = %s", $key, $confighash[$key])
}

($result << []).join("\n")
```